### PR TITLE
#184347048 refactoring public chat

### PR DESCRIPTION
**Pull request by Abdden**
### What does this PR do?

- Adds pagination to the chat functionality on the platform

### Description of Task to be completed?

- Given the user is on the chat, they should be able to see a limited number of old messages.

### How should this be manually tested?

1. Checkout to this branch: `refactor/chat-184347048`
2. Run the command `npm install` to install packages
3. Run `npm run dev`
4. Open your browser and visit `http://localhost:5000/api-docs`

### What are the relevant pivotal tracker stories?

[184347048](https://www.pivotaltracker.com/story/show/184347048)

### CODE REVIEW

**Review by irakozetony — APPROVED** [review]

**Review by ndatumuremyi — APPROVED** [review]

**Review by nshimiyejayd — APPROVED** [review]

**Review by istafrica-dennw — APPROVED** [review]

File: src/controllers/chat.controller.js [hunk 1]
```diff
@@ -20,10 +20,11 @@ export class ChatController {
 
   static async getMessages(req, res) {
     try {
-      const messages = await ChatService.getMessages();
+      const { limit, offset } = req.query;
```
Comment (TuyizeeAnastase) [inline src/controllers/chat.controller.js]: Add default limit when is empty from user

File: src/services/chat.service.js [hunk 2]
```diff
@@ -6,8 +6,11 @@ export class ChatService {
     return message;
   }
 
-  static async getMessages() {
+  static async getMessages(limit, offset) {
     const messages = await Message.findAll({
```
Comment (TuyizeeAnastase) [inline src/services/chat.service.js]: implement full pagination refer to get All products services

File: src/controllers/chat.controller.js [hunk 3]
```diff
@@ -20,10 +20,11 @@ export class ChatController {
...
+      const messages = await ChatService.getMessages(limit, offset);
       return res
         .status(200)
-        .json({ message: 'Fetched all old messages', messages });
+        .json({ message: 'Fetched old messages', messages });
```
Comment (TuyizeeAnastase) [inline src/controllers/chat.controller.js:29]: in response return total pages,currentPage,total messages and messages

**Review by TuyizeeAnastase — CHANGES_REQUESTED** [review]

**Review by TuyizeeAnastase — APPROVED** [review]

**Review by TuyizeeAnastase — CHANGES_REQUESTED** [review]
Tests aren't passing

**Review by TuyizeeAnastase — APPROVED** [review]

**Review by ncutixavier — APPROVED** [review]
LGTM

**Pull request merged**